src/operator/regression_output-inl.h

```diff
@@ -121,6 +174,80 @@ void RegressionBackward(const nnvm::NodeAttrs& attrs,
   });
 }
 
+/*!
+ * \brief Regression backward kernel for the case that
+ * label is of csr storage type.
+ * Parallelize by each row.
+ */
+template<typename OP, int req>
+struct SparseRegressionBWDKernel {
+  template<typename DType, typename IType, typename RType>
+  MSHADOW_XINLINE static void Map(int i, DType* out_data,
+                                  const DType* in_data,
+                                  const DType* label_data,
+                                  const IType* label_idx,
+                                  const RType* label_indptr,
+                                  const nnvm::dim_t row_length) {
+    nnvm::dim_t row_i = i * row_length;
+    RType k = label_indptr[i];
+    for (nnvm::dim_t j=0; j < row_length; j++) {
+      if (label_idx[k] == j) {
```

Comment:
This creates many branches and hurts performance. What about do it in 2 passes? First copy in_data, then apply minus/minus_sign only for non-zero entries?